### PR TITLE
Legacy mode for circus 2 to ease comparison with circus 1

### PR DESCRIPTION
**Pull request by yger**
Add a legacy mode to circus 2, in order to ease the comparison with circus 1. When legacy is True in the clustering, we will use a clustering very similar to the one performed by circus 1, and thus with performances that should be in line

### CODE REVIEW

**Comment by alejoe91** [comment]
Sweet! Have you made some benchmarks with SC1?

**Comment by yger** [comment]
Yes, it was the whole point, and the good news is that this is in line with SC1, giving similar results. This is a building brick for our next paper on the components :-)

**Comment by yger** [comment]
This branch is bringing quite a lot a modifications but only to the files owned and used by circus2. The key point is to have a clustering very similar to the one of circus1, such that we can properly compare everything w.r.t the new clustering of SC2. The matching engine has also been consolidated with exhaustive comparison on real recordings

File: src/spikeinterface/sortingcomponents/clustering/circus.py [hunk 1]
```diff
@@ -12,13 +12,60 @@
...
+    run_node_pipeline,
+    ExtractDenseWaveforms,
+    ExtractSparseWaveforms,
+    PeakRetriever,
+)
+
+
+def extract_waveform_at_max_channel(rec, peaks, ms_before=0.5, ms_after=1.5, **job_kwargs):
+    """
+    Helper function to extractor waveforms at max channel from a peak list
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/clustering/circus.py:37]:
```suggestion
    Helper function to extract waveforms at the max channel from a peak list
```

File: src/spikeinterface/sortingcomponents/matching/circus.py [hunk 2]
```diff
@@ -495,24 +495,22 @@ class CircusOMPSVDPeeler(BaseTemplateMatchingEngine):
...
-    noise_levels: array
-        The noise levels, for every channels. If None, they will be automatically
-        computed
-    random_chunk_kwargs: dict
-        Parameters for computing noise levels, if not provided (sub optimal)
+        Stopping criteria of the OMP algorithm, as relative error
     sparse_kwargs: dict
         Parameters to extract a sparsity mask from the waveform_extractor, if not
         already sparse.
+    rank: int
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/matching/circus.py:502]:
```suggestion
    rank: int, default: 5
```

new docstring code style

File: src/spikeinterface/sortingcomponents/matching/circus.py [hunk 3]
```diff
@@ -495,24 +495,22 @@ class CircusOMPSVDPeeler(BaseTemplateMatchingEngine):
...
+        Number of components used internally by the SVD (default 5)
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/matching/circus.py:503]:
```suggestion
        Number of components used internally by the SVD
```

File: src/spikeinterface/sortingcomponents/clustering/circus.py [hunk 4]
```diff
@@ -27,186 +74,151 @@ class CircusClustering:
...
-        params = cls._check_params(recording, peaks, params)
-        d = params
-
-        if d["peak_locations"] is None:
-            from spikeinterface.sortingcomponents.peak_localization import localize_peaks
-
-            peak_locations = localize_peaks(recording, peaks, **d["peak_localization_kwargs"], **d["job_kwargs"])
-        else:
-            peak_locations = d["peak_locations"]
+        assert HAVE_HDBSCAN, "random projections clustering need hdbscan to be installed"
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/clustering/circus.py:98]:
```suggestion
        assert HAVE_HDBSCAN, "random projections clustering needs hdbscan to be installed"
```

**Review by zm711 — COMMENTED** [review]
I just added a few docstring/assert fixes. Feel free to use if you want them.

**Comment by yger** [comment]
Thanks a lot for the comments!

File: src/spikeinterface/comparison/groundtruthstudy.py [hunk 5]
```diff
@@ -21,7 +21,7 @@
 
 
 # This is to separate names when the key are tuples when saving folders
-_key_separator = " ## "
+_key_separator = "--"
```
Comment (samuelgarcia) [inline src/spikeinterface/comparison/groundtruthstudy.py:24]:
```suggestion
_key_separator = "_##_"
```

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**